R/annotation-data-helper.R

```diff
@@ -73,25 +66,22 @@ getFirstExonRanges <- function(exonRangesByTx.unlist, transcriptRanges) {
 }
 
```

Comment:
This function merges overlapping exons for each gene. 
The new implementation does this by grouping exons by gene, and creates a variable exonClass, which assigns a unique id to groups of overlapping exons for each gene. 
We then group by this new variable, and summarize records. The new record is defined by the minimal start coordinates, and maximal end coordinate of the exons in the same exonClass. This has the same effect as merging overlapping exon regions.